### DISCUSSION TITLE
Feature Request: Don't request permissions to act on a user's behalf when signing in from Github

### DISCUSSION THREAD

**Discussion by seveibar** [tscircuit/tscircuit#459] [General]
The reason we do this is we're using Github App-style authentication rather than Github OAuth-style authentication. The former gives us a couple advantages. For development, it allows us to have a multiple callback urls for testing. For production, it allows us to easily add permission (with mandatory user consent!) to additional features, such as the ability to run workflows later.

This is a concern for users, and I'm open to changing how we do authentication to avoid this confusing permission notice, reproduced below:

![image](https://github.com/user-attachments/assets/471c7aa6-323d-48a9-be00-04def022012d)

Note: tscircuit as of the time of writing does not have the permissions or any code that actually modifies your account. We only check that you own your github email and username.

**Comment by nathaniel-brough** [comment]
This might sound discouraging, but this was a deal-breaker for me trying the online tool. All round the project looks awesome, I'll just be using the cli version.
